tools/flang1/flang1exe/semfunc.c

```diff
@@ -4264,11 +4318,25 @@ ref_intrin(SST *stktop, ITEM *list)
       } else if (argdtype == DT_WORD) {
       }
     }
+
+    if (intrin == I_MAX || intrin == I_MIN) {
+      check_max_min_argument(argdtype, &dtype_new, &dtype_last);
+    }
+
     if (!dtype1) {
       f_dt = dtype1 = argdtype; /* Save 1st arg's data type */
       if (DTY(argdtype) == TY_ARRAY)
         break;
     } else {
+      if (intrin == I_MAX || intrin == I_MIN) {
+        int dtypecompare = DTYG(argdtype);
+        if (dtypecompare == TY_WORD) dtypecompare = TY_INT;
+        if (dtypecompare == TY_DWORD) dtypecompare = TY_INT8;
+        if (dtypecompare > maxtype)
+          maxtype = dtypecompare;
+	  maxdtyppe = argdtype;
```

Comment:
Typo: `maxdtyppe` &rarr; `maxdtype`

Reply:
Also, this line is not indented correctly. It's causing a `-Wmisleading-indentation` error in the CI.

Also, I don't understand why you need two variables. It seems that you only need one.